src/wp-includes/class-wp-navigation-fallback.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * WP_Navigation_Fallback class
+ *
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @package WordPress
+ * @subpackage Navigation
+ * @since 6.3.0
+ */
+
+/**
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @access public
+ * @since 6.3.0.
+ */
+class WP_Navigation_Fallback {
+
+	/**
+	 * Gets (and/or creates) an appropriate fallback Navigation Menu.
+	 *
+	 * @since 6.3.0.
+	 *
+	 * @return WP_Post|null the fallback Navigation Post or null.
+	 */
+	public static function get_fallback() {
+
+		$fallback = static::get_most_recently_published_navigation();
+
+		if ( $fallback ) {
+			return $fallback;
+		}
+
+		$fallback = static::create_classic_menu_fallback();
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			// Return the newly created fallback post object which will now be the most recently created navigation menu.
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		$fallback = static::create_default_fallback();
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			// Return the newly created fallback post object which will now be the most recently created navigation menu.
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the most recently published `wp_navigation` post type.
+	 *
+	 * @since 6.3.0.
+	 *
+	 * @return WP_Post|null the first non-empty Navigation or null.
+	 */
+	private static function get_most_recently_published_navigation() {
+
+		$parsed_args = array(
+			'post_type'              => 'wp_navigation',
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'order'                  => 'DESC',
+			'orderby'                => 'date',
+			'post_status'            => 'publish',
+			'posts_per_page'         => 1,
+		);
+
+		$navigation_post = new WP_Query( $parsed_args );
+
+		if ( count( $navigation_post->posts ) > 0 ) {
```

Comment:
```suggestion
		if ( $navigation_post->found_posts > 0 ) {
```

Instead of count use `found_posts`

Reply:
Thanks @mukeshpanchal27 

Makes sense to me. 

Looks like this has been committed. I'll do a follow up after Beta 1.

Reply:
Core class: https://github.com/WordPress/wordpress-develop/pull/4733
Updating Gutenberg: https://github.com/WordPress/gutenberg/pull/51999

Reply:
Might have to follow up another day actually 

`$navigation_post->found_posts` returns `0` and I haven't got time to dig around. Sorry!